core/pool_vector.h

```diff
@@ -442,15 +442,50 @@ class PoolVector {
 	}
 
 	String join(String delimiter) {
-		String rs = "";
-		int s = size();
-		Read r = read();
-		for (int i = 0; i < s; i++) {
-			rs += r[i] + delimiter;
-		}
-		rs.erase(rs.length() - delimiter.length(), delimiter.length());
-		return rs;
-	}
+        int s = size();
```

Comment:
Indentation is wrong, you should use tabs.

Reply:
Thx, fixed. But I had to do additonal commits because of the flow on my side. (This is just an upstream repository.)

Reply:
I do need you to rebase your branch and squash the commits for it to be mergeable though.

But I just noticed that you made this PR against the 3.1 branch, while it should be made against the master branch.

Reply:
Okay, sure. Will do so later.

BTW: You can perform a squash merge on a PR by yourself: https://help.github.com/en/articles/configuring-commit-squashing-for-pull-requests

Reply:
I know, but we restrict the use of this feature in this repo for various reasons.